### ISSUE TITLE
Improvements

### ISSUE THREAD

**Issue by mueller-ma** [metbril/openhab-mdi#5]
Hi,

I have a few suggestions:

- Add background templates (circle, square, etc.) and a command line option, e.g. --background circle.svg
- Add command line paramter to overwrite background and icon color. When you want to have colored backgrounds it might be a good idea to set all icons to black or white without the need to change .yaml files
- After this is done we could try to add several material themes (black icons with no background, icons with circle background, etc.) to ESH or openHAB

**Comment by metbril** [comment]
Good suggestions. I'll give it some thought how to implement this, but feel free to submit one or more PR's.

If the repo becomes of more importance, we can always try to move in to the openhab organization https://github.com/openhab ;-)

**Comment by mueller-ma** [comment]
From this discussion https://github.com/openhab/openhab2-addons/issues/2139#issuecomment-331795651 it's clear that different backgrounds are not really wanted serverside in openhab, so we should forgot about this idea.

For generall color improvments:
I would change the attribute "color" to "colorState" (Color based on the item state) and add "colorRandom" ("Random" color. I use quotation marks, because these colors should be hard coded to have the same color on every run and to ensure some icons have a specific color (e.g. fire should be red)) and "colorCombined" (Use random color if no state color is present). The color used should be configurable via a cli arg.

Also I would like to add the argument "--color" which overrides all colors defined in the yaml.

**Comment by mueller-ma** [comment]
Variable names for colors instead of their hex code are IMO better readable.

**Comment by metbril** [comment]
You are correct, it makes more sense to use named colors. However, I don’t think the material design colors you also suggested have names. My initial reason for using hex codes was because the iOS app did not support them. I am unsure it does now for any arbitrary color.
